tokio/src/net/tcp/socket.rs

```diff
@@ -482,6 +482,48 @@ impl TcpSocket {
         let mio = self.inner.listen(backlog)?;
         TcpListener::new(mio)
     }
+
+    /// Converts a TcpStream into a TcpSocket
+    /// This converts the TcpStream into
+    /// the platform specific raw_fd(unix) / raw_socket(windows)
+    /// and then converts them into the TcpSocket
+    /// as TcpSocket implements the FromRawFd(unix) and
+    /// FromRawSocket(windows) traits.
+
```

Comment:
I have some improvements to the documentation:

 * We typically wrap the text such that it goes all the way out to 80 characters wide. If you want to make a newline that shows up in the documentation, you need an empty line, as it otherwise just concatenates the lines.
 * I would like `TcpStream` to be a link, and `TcpSocket` should be surrounded by backticks to display it as code.
 * You do not have to explain _how_ it does the conversion.
 * The documentation should make it clear that this is intended for usage together with `socket2`, and that the socket must not be in a connected state.

Reply:
Thanks for your quick reply. I'll go over all the changes suggested and raise another request.

Reply:
You should not create a new PR. Just push more commits to your `tcp-socket-from-std-stream` branch.